core/src/service/path_service.rs

```diff
@@ -148,6 +149,27 @@ pub fn get_all_paths(config: &Config, filter: Option<Filter>) -> Result<Vec<Stri
     Ok(paths)
 }
 
+pub fn get_path_by_id(config: &Config, id: Uuid) -> Result<String, CoreError> {
+    let mut current_id = id;
+    let mut current_metadata = file_metadata_repo::get(config, current_id)?;
+    let mut path = String::from("");
+
+    while current_metadata.parent != current_id {
+        let name = file_encryption_service::get_name(&config, &current_metadata)?;
+        path = format!("{}/{}", name, path);
+        current_id = current_metadata.parent;
+        current_metadata = file_metadata_repo::get(config, current_id)?
+    }
+
+    {
+        let name = file_encryption_service::get_name(&config, &current_metadata)?;
+        path = format!("{}/{}", name, path);
+    }
```

Comment:
maybe you'd be better off modeling it this way: https://github.com/rust-lang/rfcs/issues/1313

Reply:
But I want to check the condition immediately